codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/parse/JsonParserGenerator.kt

```diff
@@ -560,6 +562,15 @@ class JsonParserGenerator(
                             *codegenScope,
                         ) {
                             objectKeyLoop(hasMembers = shape.members().isNotEmpty()) {
+                                rustTemplate(
+                                    """
+                                    if let #{Some}(#{Ok}(#{Token}::ValueNull { .. })) = tokens.peek() {
+                                        #{skip_value}(tokens)?;
```

Comment:
Calling `skip_value` seems excessive if you already know it's null. Should just be able to `pop` it after.